stackdriver/client_test.go

```diff
@@ -114,3 +117,43 @@ func TestEmptyRequest(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestResolver(t *testing.T) {
+  grpcServer := grpc.NewServer()
+	listener := newLocalListener()
+	go grpcServer.Serve(listener)
+	defer grpcServer.Stop()
+
+	serverURL, err := url.Parse("http://stackdriver.invalid?auth=false")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	res, _ := manual.GenerateAndRegisterManualResolver()
+	res.InitialAddrs([]resolver.Address{
+		{Addr: "localhost"},
+	})
+	c := NewClient(&ClientConfig{
+		URL:      serverURL,
+		Timeout:  time.Second,
+		Resolver: res,
+	})
+
+  address := c.url.Hostname()
+	ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+	defer cancel()
+
+	c.resolver.Scheme()
+
+	conn, connerr := grpc.DialContext(ctx, address, grpc.WithInsecure())
+	c.conn = conn
+	if connerr != nil {
```

Comment:
Nit, can you see if the above works written as follows? I want to avoid the extra variables and assignments, as they could cause errors.

`if c.conn, err = grpc.DialContext(ctx, address, grpc.WithInsecure()); err  != nil {`